Realm/Realm/DatabaseTypes/RealmDictionary.cs

```diff
@@ -98,6 +98,12 @@ public ICollection<TValue> Values
             }
         }
 
+        internal RealmResults<TValue> ToResults()
+        {
+            var resultsHandle = _dictionaryHandle.GetValues();
+            return new RealmResults<TValue>(Realm, resultsHandle, Metadata);
```

Comment:
it could have been nice to just do i.e. `this.Values` but that would involve double-casting to go back to `RealmResults` from `Collection`, I am not sure if that's a good idea